### ISSUE TITLE
Ubuntu 12.04 Compile error: clang++-3.5: not found

### ISSUE THREAD

**Issue by superclarkk** [BartVandewoestyne/oaconvert#134]
Any advice to resolve this issue would be appreciated

Environment:
* Ubuntu 12.04
* `Linux myhost 3.2.0-83-generic #120-Ubuntu SMP Wed Apr 29 15:37:05 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

```
root@myhost:/usr/local/bin/3rdparty/oaconvert# VER=4.6 ; PRIO=40
root@myhost:/usr/local/bin/3rdparty/oaconvert# update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$VER $PRIO --slave /usr/bin/g++ g++ /usr/bin/g++-$VER
root@myhost:/usr/local/bin/3rdparty/oaconvert# update-alternatives --install /usr/bin/cpp cpp-bin /usr/bin/cpp-$VER $PRIO
root@myhost:/usr/local/bin/3rdparty/oaconvert# VER=4.7 ; PRIO=60
root@myhost:/usr/local/bin/3rdparty/oaconvert# update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$VER $PRIO --slave /usr/bin/g++ g++ /usr/bin/g++-$VER
update-alternatives: using /usr/bin/gcc-4.7 to provide /usr/bin/gcc (gcc) in auto mode.
root@myhost:/usr/local/bin/3rdparty/oaconvert# update-alternatives --install /usr/bin/cpp cpp-bin /usr/bin/cpp-$VER $PRIO
update-alternatives: using /usr/bin/cpp-4.7 to provide /usr/bin/cpp (cpp-bin) in auto mode.
root@myhost:/usr/local/bin/3rdparty/oaconvert# g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.7/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 4.7.3-2ubuntu1~12.04' --with-bugurl=file:///usr/share/doc/gcc-4.7/README.Bugs --enable-languages=c,c++,go,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.7 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.7 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --with-system-zlib --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.7.3 (Ubuntu/Linaro 4.7.3-2ubuntu1~12.04) 
root@myhost:/usr/local/bin/3rdparty/oaconvert# rm CMakeCache.txt 
root@myhost:/usr/local/bin/3rdparty/oaconvert# cmake .
-- The C compiler identification is GNU 4.7.3
-- The CXX compiler identification is GNU 4.7.3
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/gcc-4.7
-- Check for working CXX compiler: /usr/bin/gcc-4.7 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/local/bin/3rdparty/oaconvert
root@myhost:/usr/local/bin/3rdparty/oaconvert# cd src/
root@myhost:/usr/local/bin/3rdparty/oaconvert/src# make
[  5%] Building CXX object src/CMakeFiles/airspace.dir/Airspace.cpp.o
/bin/sh: 1: /usr/bin/clang++-3.5: not found
make[2]: *** [src/CMakeFiles/airspace.dir/Airspace.cpp.o] Error 127
make[1]: *** [src/CMakeFiles/airspace.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by superclarkk** [comment]
Note: terminal output in previous post has been edited. Now using gcc version 4.7.3

**Comment by superclarkk** [comment]
Note that I have exactly the same error on OS X 10.9.5

**Comment by Trogie** [comment]
On 09-08-15 17:40, superclarkk wrote:
> Any advice to resolve this issue would be appreciated
>
> Environment:
>
>   * Ubuntu 12.04
>   * |Linux myhost 3.2.0-83-generic #120-Ubuntu SMP Wed Apr 29 15:37:05
>     UTC 2015 x86_64 x86_64 x86_64 GNU/Linux|
>
> |root@myhost:/usr/local/bin/3rdparty/oaconvert# cmake .
> -- The C compiler identification is GNU 4.6.4
> -- The CXX compiler identification is GNU 4.6.4
> -- Check for working C compiler: /usr/bin/gcc
> -- Check for working C compiler: /usr/bin/gcc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /usr/local/bin/3rdparty/oaconvert
> root@myhost:/usr/local/bin/3rdparty/oaconvert# cd src/
> root@myhost:/usr/local/bin/3rdparty/oaconvert/src# make
> Scanning dependencies of target airspace
> [  5%] Building CXX object src/CMakeFiles/airspace.dir/Airspace.cpp.o
> /bin/sh: 1: /usr/bin/clang++-3.5: not found
> make[2]: *** [src/CMakeFiles/airspace.dir/Airspace.cpp.o] Error 127
> make[1]: *** [src/CMakeFiles/airspace.dir/all] Error 2
> make: *** [all] Error 2
> root@myhost:/usr/local/bin/3rdparty/oaconvert/src# make VERBOSE=1
> cd /usr/local/bin/3rdparty/oaconvert && /usr/bin/cmake -H/usr/local/bin/3rdparty/oaconvert -B/usr/local/bin/3rdparty/oaconvert --check-build-system CMakeFiles/Makefile.cmake 0
> cd /usr/local/bin/3rdparty/oaconvert && /usr/bin/cmake -E cmake_progress_start /usr/local/bin/3rdparty/oaconvert/CMakeFiles /usr/local/bin/3rdparty/oaconvert/src/CMakeFiles/progress.marks
> cd /usr/local/bin/3rdparty/oaconvert && make -f CMakeFiles/Makefile2 src/all
> make[1]: Entering directory `/usr/local/bin/3rdparty/oaconvert'
> make -f src/CMakeFiles/airspace.dir/build.make src/CMakeFiles/airspace.dir/depend
> make[2]: Entering directory `/usr/local/bin/3rdparty/oaconvert'
> cd /usr/local/bin/3rdparty/oaconvert && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /usr/local/bin/3rdparty/oaconvert /usr/local/bin/3rdparty/oaconvert/src /usr/local/bin/3rdparty/oaconvert /usr/local/bin/3rdparty/oaconvert/src /usr/local/bin/3rdparty/oaconvert/src/CMakeFiles/airspace.dir/DependInfo.cmake --color=
> make[2]: Leaving directory `/usr/local/bin/3rdparty/oaconvert'
> make -f src/CMakeFiles/airspace.dir/build.make src/CMakeFiles/airspace.dir/build
> make[2]: Entering directory `/usr/local/bin/3rdparty/oaconvert'
> /usr/bin/cmake -E cmake_progress_report /usr/local/bin/3rdparty/oaconvert/CMakeFiles 9
> [  5%] Building CXX object src/CMakeFiles/airspace.dir/Airspace.cpp.o
> cd /usr/local/bin/3rdparty/oaconvert/src && /usr/bin/clang++-3.5   -Dairspace_EXPORTS -std=c++14 -stdlib=libc++ -g -fPIC   -o CMakeFiles/airspace.dir/Airspace.cpp.o -c /usr/local/bin/3rdparty/oaconvert/src/Airspace.cpp
> /bin/sh: 1: /usr/bin/clang++-3.5: not found
> make[2]: *** [src/CMakeFiles/airspace.dir/Airspace.cpp.o] Error 127
> make[2]: Leaving directory `/usr/local/bin/3rdparty/oaconvert'
> make[1]: *** [src/CMakeFiles/airspace.dir/all] Error 2
> make[1]: Leaving directory `/usr/local/bin/3rdparty/oaconvert'
> make: *** [all] Error 2
> |
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BartVandewoestyne/oaconvert/issues/134>.
>


1. Your ubuntu version is pretty old and not supported anymore
2. Install the clang package and not only the clang-x.x packages.

Tried it on my working laptop where I didn't have clang package 
installed. After installation of the dependency packages and clang I 
have the /usr/bin/clang++ in my path.

Kind regards,
Tom Rogie

**Comment by superclarkk** [comment]
As clang-3.5 is not available for 12.04, and that my system works perfectly for all my other purposes and so I don't intend to upgrade for this, I'm going to abandon trying to get it to work on this system. I'm closing this ticket.

I've opened #135 to get it running on OS X 10.9.5 instead.
